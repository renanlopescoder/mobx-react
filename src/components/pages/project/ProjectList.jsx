import React from 'react'
import { inject, observer } from 'mobx-react'
import { Link } from 'react-router-dom'
import { Grid } from 'react-flexbox-grid'
import {
  CardTitle,
  CardText,
  Table,
  TableHead,
  TableCell,
  TableRow,
  IconButton
} from 'react-toolbox'

import ProjectSearchForm from './ProjectSearchForm'
import { Loader, Card, AddButton } from '../../atoms'
import color from '../../../shared/colors.css'

@inject('projectStore')
@observer
class ProjectList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      query: {}
    };
    props.projectStore.fetchProjectList();
  };

  getTableHead = () => (
    <TableHead>
      <TableCell>ID</TableCell>
      <TableCell>Name</TableCell>
      <TableCell>Description</TableCell>
      <TableCell>GitHub Link</TableCell>
      <TableCell>DEMO Link</TableCell>
      <TableCell>Technologies</TableCell>
      <TableCell>Status</TableCell>
      <TableCell>Author</TableCell>
      <TableCell>Author Link</TableCell>
      <TableCell>Actions</TableCell>
    </TableHead>
  );

  getTableBody = project => (
    <TableRow key={project._id}>
      <TableCell>{project._id}</TableCell>
      <TableCell>{project.project}</TableCell>
      <TableCell>{project.description}</TableCell>
      <TableCell>{project.githubLink}</TableCell>
      <TableCell>{project.demoLink}</TableCell>
      <TableCell>{project.technologies}</TableCell>
      <TableCell>{project.status}</TableCell>
      <TableCell>{project.author}</TableCell>
      <TableCell>{project.authorLink}</TableCell>
      <TableCell>
        <Link to={`/project/form/${project._id}`}>
          <IconButton className={color.secondaryText} icon='edit'  />
        </Link>
        <IconButton style={{color: '#9A338E'}} icon='delete' onClick={() => this.props.projectStore.deleteProject(project._id)}  />
      </TableCell>
    </TableRow>
  );

  search = query => {
    this.setState({query});
  };

  render() {
    return (
      <Grid fluid>
        <ProjectSearchForm search={this.search} />
        <Card>
          <CardTitle title='Project List' />
          <CardText>
            <Loader loading={this.props.projectStore.loading}>
              <Table selectable={false}>
                {this.getTableHead()}
                {this.props.projectStore.projectList(this.state.query).map(project => (
                  this.getTableBody(project)
                ))}
              </Table>
              <AddButton to='/project/form' />
            </Loader>
          </CardText>
        </Card>
      </Grid>
    );
  };
};

export default ProjectList;
