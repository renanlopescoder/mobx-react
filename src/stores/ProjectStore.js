import { types } from "mobx-state-tree";

import Project from './models/Project';
import ProjectActions from './actions/ProjectActions';
import ProjectViews from './views/ProjectViews';

const ProjectStore = types.model('ProjectStore', {
  loading: types.maybe(types.boolean),
  projects: types.map(Project),
})
.views(ProjectViews)
.actions(ProjectActions);


export default ProjectStore;
