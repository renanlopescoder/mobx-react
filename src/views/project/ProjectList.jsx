import React from 'react';
import { inject, observer } from 'mobx-react';
import { Grid } from 'react-flexbox-grid';
import { CardTitle, CardText, Table, TableHead, TableCell, TableRow } from 'react-toolbox';

import { Loader, Card, AddButton } from '../../components';

@inject('projectStore')
@observer
class ProjectList extends React.Component {
  constructor(props) {
    super(props);
    props.projectStore.fetchProjectList();
  };

  getTableHead = () => (
    <TableHead>
      <TableCell>ID</TableCell>
      <TableCell>Name</TableCell>
      <TableCell>Description</TableCell>
      <TableCell>GitHub Link</TableCell>
      <TableCell>Author</TableCell>
    </TableHead>
  );

  getTableBody = project => (
    <TableRow key={project._id}>
      <TableCell>{project._id}</TableCell>
      <TableCell>{project.project}</TableCell>
      <TableCell>{project.description}</TableCell>
      <TableCell>{project.githubLink}</TableCell>
      <TableCell>{project.author}</TableCell>
    </TableRow>
  );

  render() {
    return (
      <Grid fluid>
        <Card>
          <CardTitle title={`Project List`} />
          <CardText>
            <Loader loading={this.props.projectStore.loading}>
              <Table selectable={false}>
                {this.getTableHead()}
                {this.props.projectStore.projects.values().map(project => (
                  this.getTableBody(project)
                ))}
              </Table>
              <AddButton to='/projects/new'/>
            </Loader>
          </CardText>
        </Card>
      </Grid>
    );
  };
};

export default ProjectList;
