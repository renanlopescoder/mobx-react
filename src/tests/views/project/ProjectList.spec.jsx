import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

import ProjectStore from '../../../stores/ProjectStore'
import ProjectList from '../../../views/project/ProjectList'

describe('view/<ProjectList />', () => {
  const projectStore = ProjectStore.create()
  it('should render the question header', () => {
    const component = renderer.create(
      <MemoryRouter>
        <ProjectList projectStore={projectStore} />
      </MemoryRouter>,
    )
    const json = component.toJSON()
    expect(json).toMatchSnapshot()
  })
})
