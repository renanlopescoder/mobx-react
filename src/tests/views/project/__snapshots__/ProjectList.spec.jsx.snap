// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view/<ProjectList /> should render the question header 1`] = `
<div
  className="container-fluid"
>
  <div
    className=""
    data-react-toolbox="card"
  >
    <div
      className="undefined"
    >
      <div>
        <h5
          className={undefined}
        >
          Project List
        </h5>
      </div>
    </div>
    <div
      className=""
    >
      <div
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="undefined"
        data-react-toolbox="progress-bar"
        disabled={undefined}
      >
        <svg
          className={undefined}
          viewBox="0 0 60 60"
        >
          <circle
            className={undefined}
            cx="30"
            cy="30"
            r="25"
            style={undefined}
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
